tsp-output/@azure-tools/typespec-csharp/src/Generated/Audio.cs

```diff
@@ -331,91 +328,89 @@ public virtual async Task<Result> CreateTranslationAsync(RequestBody content, Re
-        public virtual Result CreateTranslation(RequestBody content, RequestOptions context = null)
+        public virtual ClientResult CreateTranslation(BinaryContent content, string contentType, RequestOptions options = null)
         {
-            ClientUtilities.AssertNotNull(content, nameof(content));
+            Argument.AssertNotNull(content, nameof(content));
 
-            using var scope = ClientDiagnostics.CreateSpan("Audio.CreateTranslation");
-            scope.Start();
-            try
-            {
-                using PipelineMessage message = CreateCreateTranslationRequest(content, context);
-                return Result.FromResponse(_pipeline.ProcessMessage(message, context));
-            }
-            catch (Exception e)
+            options ??= new RequestOptions();
+
+            using PipelineMessage message = CreateCreateTranslationRequest(content, contentType, options);
+
+            _pipeline.Send(message);
+
+            PipelineResponse response = message.Response!;
+
+            if (response.IsError && options.ErrorOptions == ClientErrorBehaviors.Default)
             {
-                scope.Failed(e);
-                throw;
+                throw new ClientResultException(response);
             }
+
+            return ClientResult.FromResponse(response);
         }
 
-        internal PipelineMessage CreateCreateSpeechRequest(RequestBody content, RequestOptions context)
+        internal PipelineMessage CreateCreateSpeechRequest(BinaryContent content, RequestOptions options)
         {
-            var message = _pipeline.CreateMessage(context, ResponseErrorClassifier200);
+            var message = _pipeline.CreateMessage();
+            message.ResponseClassifier = PipelineMessageClassifier.Create(stackalloc ushort[] { 200 });
             var request = message.Request;
-            request.SetMethod("POST");
-            var uri = new RequestUri();
-            uri.Reset(_endpoint);
-            uri.AppendPath("/audio/speech", false);
-            request.Uri = uri.ToUri();
-            request.SetHeaderValue("Accept", "application/octet-stream");
-            request.SetHeaderValue("Content-Type", "application/json");
+            request.Method = "POST";
+            var uriBuilder = new UriBuilder(_endpoint.ToString());
+            var path = new StringBuilder();
+            path.Append("/audio/speech");
+            uriBuilder.Path = path.ToString();
+            request.Uri = uriBuilder.Uri;
+            request.Headers.Set("Accept", "application/octet-stream");
+            request.Headers.Set("Content-Type", "application/json");
             request.Content = content;
+            message.Apply(options);
             return message;
         }
 
-        internal PipelineMessage CreateCreateTranscriptionRequest(RequestBody content, RequestOptions context)
+        internal PipelineMessage CreateCreateTranscriptionRequest(BinaryContent content, string contentType, RequestOptions options)
```

Comment:
Added contentType value as an input parameter based on conversation in Azure.Core meeting.